ddl/ddl_api.go

```diff
@@ -1278,38 +1279,35 @@ func (d *ddl) CreateView(ctx sessionctx.Context, s *ast.CreateViewStmt) (err err
 	return d.callHookOnChanged(err)
 }
 
-func buildViewInfoWithTableColumns(ctx sessionctx.Context, s *ast.CreateViewStmt) (*model.ViewInfo, []*table.Column) {
+func buildViewInfoWithTableColumns(ctx sessionctx.Context, s *ast.CreateViewStmt) (*model.ViewInfo, []*table.Column, error) {
```

Comment:
We can split this function into 2 parts now.
1. s/ buildViewInfoWithTableColumns/ buildViewInfo
2. extract the `buildTableColumns` part to line 1242 instead wrapping it as a function

Reply:
fine